fe/fe-core/src/main/java/com/starrocks/execution/DropFunctionExecutor.java

```diff
@@ -3,15 +3,36 @@
 package com.starrocks.execution;
 
 import com.starrocks.analysis.DropFunctionStmt;
+import com.starrocks.analysis.FunctionName;
 import com.starrocks.analysis.StatementBase;
+import com.starrocks.catalog.Database;
+import com.starrocks.catalog.FunctionSearchDesc;
+import com.starrocks.common.ErrorCode;
+import com.starrocks.common.ErrorReport;
 import com.starrocks.common.UserException;
 import com.starrocks.qe.ConnectContext;
 import com.starrocks.qe.ShowResultSet;
+import com.starrocks.server.GlobalStateMgr;
 
 public class DropFunctionExecutor implements DataDefinitionExecutor {
 
     public ShowResultSet execute(StatementBase stmt, ConnectContext context) throws UserException {
-        context.getGlobalStateMgr().dropFunction((DropFunctionStmt) stmt);
+        DropFunctionStmt dropFunctionStmt = (DropFunctionStmt) stmt;
+        FunctionName name = dropFunctionStmt.getFunctionName();
+        Database db = context.getGlobalStateMgr().getDb(name.getDb());
+        if (db == null) {
+            ErrorReport.reportDdlException(ErrorCode.ERR_BAD_DB_ERROR, name.getDb());
+        }
+        db.dropFunction(dropFunctionStmt.getFunction());
         return null;
     }
+
+    public static void replayDropFunction(FunctionSearchDesc functionSearchDesc) {
```

Comment:
This replay logic should be put in Database.java

Reply:
ok，I will move it